### PR TITLE
add ability to set timeout upon calling connect method

### PR DESCRIPTION
**Pull request by jakerockland**

### CODE REVIEW

**Comment by jakerockland** [comment]
Currently it doesn't seem like there is a way to get around the default of 10s.

File: SwiftyBluetooth/Source/Peripheral.swift [hunk 1]
```diff
@@ -103,8 +103,8 @@ extension Peripheral {
     }
     
     /// Connect to the peripheral through Ble to our Central sharedInstance
-    public func connect(completion: @escaping ConnectPeripheralCallback) {
-        self.peripheralProxy.connect(completion)
+    public func connect(timeout: TimeInterval = 10, completion: @escaping ConnectPeripheralCallback) {
```
Comment (jordanebelanger) [inline SwiftyBluetooth/Source/Peripheral.swift]: Very good, not all swift devs use default value for parameters like this and many of them still write duplicate functions instead.

I would change it slightly to: "connect(withTimeout timeout: TimeInterval = 10"

**Comment by jordanebelanger** [comment]
Yes I never got around to figuring a proper solution for timeout customizations. I thought about making it a static/instance property of the peripheral class but that felt kinda dirty. I also wanted to avoid having a timeout parameter in all the public functions. 

I think being able to customize the connect timeout, while having the other timeouts be fixed is a good compromise.

**Comment by jakerockland** [comment]
Having it be an instance property does seem a bit tricky for the end user--not super intuitive. Definitely agree that having it in all of the public functions is a bit messy though. I currently can't think of good reasons why one would want to specify an alternative timeout for anything other than connection leaving the others fixed makes sense to me too. Just got a chance to test this out and it seems to be working appropriately 😄

**Comment by jakerockland** [comment]
However, I also could see an argument for just adding it to all public methods as an optional parameter, as it allows any of them to be called with or without the `timeout` parameter, so it doesn't actually end up making it more messy or complicated for the end user of the framework.

tl;dr - I'm not sure I see a compelling reason to *not* add the timeout parameter to all the public functions, but I also don't see a compelling reason to add it at this time, other than that it would add consistency.

I'm happy to add the timeout to other public methods if you'd like?

**Comment by jordanebelanger** [comment]
Only reason tbh is that xcode is garbage and the autocomplete for swift spew a bunch of garbage whenever you start adding parameters with default value in functions, and since the method signature are so long, you end up being unable to see which version of the method you want to choose because the signature are cropped "..."

**Comment by jakerockland** [comment]
Haha that's fair, so no need to add the timeout to other public methods eh?

**Comment by jordanebelanger** [comment]
Yep, I think timeout on connect only is good best.

**Comment by jakerockland** [comment]
Sounds good to me!

**Comment by jordanebelanger** [comment]
Please change your method signature to connect(withTimeout timeout: so I can integrate.

**Comment by jakerockland** [comment]
@tehjord Just made that change 👍
